src/sapling_sighash_versioning.rs

```diff
@@ -0,0 +1,57 @@
+//! This module defines the versioning for Sapling signatures.
+
+use redjubjub::{Binding, SigType, Signature, SpendAuth};
+
+/// The Sapling Sighash version.
+/// Represented as a `u8` for compatibility with the PCZT encoding.
+#[repr(u8)]
+#[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord)]
+pub enum SaplingSighashVersion {
+    /// Version V0.
+    V0 = 0,
+
+    /// No version (used for Sapling and TXv5 compatibility).
+    /// TXv5 does not require the sighash versioning bytes.
+    NoVersion = u8::MAX,
```

Comment:
this looks unused.
Also, can you brifly explain the backward compatibility stratagy here?

Reply:
The NoVersion is used in librustzcash when we read V3/V4/V5 transactions.

The backward compatibility strategy is only done in librustzcash.
In sapling repo, all signatures (spend auth and binding) have a version.
In librustzcash, we manage backward compatibility by 
- not writing signature version only for V6 transactions
- filling the version with NoVersion when reading signatures in V3/V4/V5 transactions